bundles/binding/org.openhab.binding.panasonictv/src/main/java/org/openhab/binding/panasonictv/internal/PanasonicTVBinding.java

```diff
@@ -0,0 +1,238 @@
+				refreshInterval = Long.parseLong(refreshIntervalString);
+			}
+
+			for (Enumeration<?> e = config.keys(); e.hasMoreElements();) {
+				String tv = (String) e.nextElement();
+				if (tv.equalsIgnoreCase("service.pid")
+						|| tv.equalsIgnoreCase("refresh"))
+					continue;
+				logger.info("TV registered '" + tv + "' with IP '"
+						+ config.get(tv) + "'");
+				registeredTVs.put(tv, config.get(tv).toString());
+			}
+
+			if (registeredTVs.isEmpty()) {
+				setProperlyConfigured(true);
+				logger.debug("No TV was registered in config file");
+			} else
+				setProperlyConfigured(true);
+		}
+	}
+
+	/**
+	 * @author André Heuer
+	 *
+	 * This methods sends the command to the TV
+	 * 
+	 * @return HTTP response code from the TV (should be 200)
+	 */
+	int sendCommand(PanasonicTVBindingConfig config) {
+		String command = config.getCommand().toUpperCase();
+
+		String soaprequest = "";
+
+		if (config.getCommand().toUpperCase().startsWith("HDMI"))
+			soaprequest = "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\"><s:Body><u:X_SendKey xmlns:u=\"urn:panasonic-com:service:p00NetworkControl:1\"><X_KeyEvent>NRC_"
+					+ command
+					+ "</X_KeyEvent></u:X_SendKey></s:Body></s:Envelope>\r";
+		else
+			soaprequest = "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\"><s:Body><u:X_SendKey xmlns:u=\"urn:panasonic-com:service:p00NetworkControl:1\"><X_KeyEvent>NRC_"
+					+ command
+					+ "-ONOFF</X_KeyEvent></u:X_SendKey></s:Body></s:Envelope>\r";
+
+		String tvIp = registeredTVs.get(config.getTv());
+
+		if ((tvIp == null) || tvIp.isEmpty())
+			return 0;
+
+		int port = 55000;
+
+		try {
+			Socket client = new Socket(tvIp, port);
+
+			BufferedWriter wr = new BufferedWriter(new OutputStreamWriter(
+					client.getOutputStream(), "UTF8"));
+
+			String header = "POST /nrc/control_0/ HTTP/1.1\r\n";
+			header = header + "Host: " + tvIp + ":55000\r\n";
+			header = header
+					+ "SOAPACTION: \"urn:panasonic-com:service:p00NetworkControl:1#X_SendKey\"\r\n";
+			header = header + "Content-Type: text/xml; charset=\"utf-8\"\r\n";
+			header = header + "Content-Length: " + soaprequest.length()
+					+ "\r\n";
+			header = header + "\r\n";
+
+			String request = header + soaprequest;
+
+			logger.debug("Request send to TV with IP " + tvIp + ": " + request);
+
+			wr.write(header);
+			wr.write(soaprequest);
+
+			wr.flush();
+
+			InputStream inFromServer = client.getInputStream();
+
+			BufferedReader reader = new BufferedReader(new InputStreamReader(
+					inFromServer));
+
+			String response = reader.readLine();
+
+			client.close();
+
+			logger.debug("TV Response from " + tvIp + ": " + response);
+
+			return Integer.parseInt(response.split(" ")[1]);
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (Exception e) {
+			logger.error("Exception in binding during execution of command: "
+					+ e.getStackTrace());
+		}
+		return 0;
+	}
+
+	@Override
+	protected void execute() {
+
+	}
```

Comment:
since you don't use polling please remove execute() and extends AbstractBinding rather than AbstractActiveBinding